internal/abci/abci.go

```diff
@@ -269,6 +289,7 @@ func NewAbciApp(ctx context.Context, cfg *AbciConfig, snapshotter SnapshotModule
 			VoteExpiry:       app.consensusParams.Votes.VoteExpiry,
 			DisabledGasCosts: app.consensusParams.WithoutGasCosts,
 			MaxVotesPerTx:    app.consensusParams.Votes.MaxVotesPerTx,
+			MigrationStatus:  status,
```

Comment:
This entire `common.NetworkParameters` is explicitly instantiated twice.  Once for `meta.StoreParams` above and then here for the `common.ChainContext` return.  This is error prone and it should be done once just before attempting `meta.StoreParams`